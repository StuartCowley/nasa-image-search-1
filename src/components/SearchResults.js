import React from 'react';
import '../styles/SearchResults.css';
import PropTypes from 'prop-types';

const SearchResults = ({ results, error }) => {
	if (error) {
		return <p>No results</p>;
	}
	if (results.length > 0) {
		return (
			<div className="search-results">
				<p>Search results</p>
				<div className="image-grid">
					{results.map(image => (
						<img
							key={image}
							className="card-image"
							src={image}
							alt="moon"
						/>
					))}
				</div>
			</div>
		);
	}
	return null;
};
export default SearchResults;

SearchResults.propTypes = {
	results: PropTypes.arrayOf(PropTypes.string),
};
