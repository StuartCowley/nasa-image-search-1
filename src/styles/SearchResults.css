.image-grid {
	display: grid;
	grid-gap: 10px;
	width: 100%;
	grid-template-columns: 1fr 1fr 1fr 1fr;
	grid-template-rows: auto;
}
.card-image {
	width: 100%;
	height: 100%;
	object-fit: cover;
}
/* .search-results {
	display: flex;
	flex-direction: row;
	flex-wrap: wrap;
	flex-shrink: initial;
	padding: 10px;
} */
