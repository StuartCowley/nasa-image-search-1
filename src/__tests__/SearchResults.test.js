import React from 'react';
import { render } from '@testing-library/react';
import SearchResults from '../components/SearchResults';

describe('SearchResults', () => {
	const validProps = {
		results: ['image.jpg', 'image2.jpg', 'image3.jpg'],
	};
	it('renders correctly', () => {
		const { asFragment } = render(
			<SearchResults results={validProps.results} />,
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
