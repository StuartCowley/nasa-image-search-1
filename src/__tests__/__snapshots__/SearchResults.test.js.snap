// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-results"
  >
    <p>
      Search results
    </p>
    <div
      class="image-grid"
    >
      <img
        alt="moon"
        class="card-image"
        src="image.jpg"
      />
      <img
        alt="moon"
        class="card-image"
        src="image2.jpg"
      />
      <img
        alt="moon"
        class="card-image"
        src="image3.jpg"
      />
    </div>
  </div>
</DocumentFragment>
`;
